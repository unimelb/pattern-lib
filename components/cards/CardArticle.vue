<template>
  <div
    class="article-card">
    <div class="article-card__inner border-top">
      <h3 class="article-card__title heading-sm heading-sm--link">
        <a
          class="article-card__link link-reset"
          :href="href">
          {{ title }}
        </a>
      </h3>
      <p class="article-card__category">
        {{ category }}
      </p>
    </div>
  </div>
</template>

<script>
export default {
  props: {
    href: {
      type: String,
      required: true,
      default: '',
    },
    title: {
      type: String,
      required: true,
      default: '',
    },
    category: {
      type: String,
      required: true,
      default: '',
    },
  },
};
</script>
