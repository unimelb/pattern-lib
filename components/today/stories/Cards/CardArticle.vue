<template>
  <CardArticle
    href="https://www.unimelb.edu.au/"
    title="Test title"
    category="Test category" />
</template>

<script>
import CardArticle from 'components/cards/CardArticle.vue';

export default {
  components: {
    CardArticle,
  },
  readme: {
    source: true,
    html: true,
  },
};
</script>
