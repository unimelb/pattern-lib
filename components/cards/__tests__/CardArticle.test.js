import {
  mount,
} from 'vue-test-utils';
import { axe, toHaveNoViolations } from 'jest-axe';
import CardArticle from '../CardArticle.vue';

expect.extend(toHaveNoViolations);

describe('CardArticle', () => {
  it('should match snapshot', () => {
    const result = mount(CardArticle, {
      propsData: {
        title: 'mock title',
        href: 'www.unimelb.edu.au',
        category: 'mock category',
      },
    }).element;
    expect(result).toMatchSnapshot();
  });

  it('Component throws no accessibility violations', (done) => {
    const html = mount(CardArticle, {
      propsData: {
        title: 'mock title',
        href: 'www.unimelb.edu.au',
        category: 'mock category',
      },
    }).html();
    // pass anything that outputs html to axe
    return axe(html).then((response) => {
      expect(response).toHaveNoViolations();
      done();
    });
  });
});
