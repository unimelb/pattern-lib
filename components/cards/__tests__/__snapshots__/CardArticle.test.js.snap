// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardArticle should match snapshot 1`] = `
<div
  class="article-card"
>
  <div
    class="article-card__inner border-top"
  >
    <h3
      class="article-card__title heading-sm"
    >
      <a
        class="article-card__link link-reset"
        href="www.unimelb.edu.au"
      >
        
        mock title
      
      </a>
    </h3>
     
    <p
      class="article-card__category"
    >
      
      mock category
    
    </p>
  </div>
</div>
`;
